src/Documentation/Event-Sourcing/MultiVersion.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: page
+title: Immediate vs. Delayed Confirmation
+---
+
+# Immediate Confirmation
+
+For many applications, we want to ensure that events are confirmed immediately, so that the persisted version does not lag behind the current version in memory, and we do not risk losing the latest state if the grain should fail. We can guarantee this by following these rules:
+
+1. Confirm all `RaiseEvent` calls using `ConfirmEvents` before the grain method returns.
+
+1. Make sure tasks returned by `RaiseConditionalEvent` complete before the grain method returns.
+
+1. Avoid  `[Reentrant]` or `[AlwaysInterleave]` attributes, so only one grain call can be processed at a time.
+
+If we follow these rules, it means that after an event is raised, no other grain code can execute until the event has been written to storage. Therefore, it is impossible to observe inconsistencies between the version in memory and the version in storage. While this is often exactly what we want, it also has some decided disadvantages.
+
+* if the connection to a remote cluster or to storage is temporarily interrupted, then the grain becomes unavailable: effectively, the grain cannot execute any code while it is stuck waiting to confirm the events, which can take an indefinite amount of time (the confirmation protocol never times out).
```

Comment:
why wouldn't it time out? This is access to storage (at least in the ES scenario), there should be a way to timeout...